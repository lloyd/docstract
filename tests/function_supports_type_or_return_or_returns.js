/**
 * A module with the three different ways to delcare
 * return types.
 */

/**
 * @function toFilename
 * a function which converts URLs to filenames
 * @returns {string} A string.
 */

/**
 * @function toFilename2
 * a function which converts URLs to filenames
 * @return {string} A String.
 */

/**
 * @function toFilename3
 * a function which converts URLs to filenames
 * @type {string} Yeah, A String.
 */
