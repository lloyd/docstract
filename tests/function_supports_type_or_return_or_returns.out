{
  "desc": "A module with the three different ways to delcare\nreturn types.", 
  "filename": "tests/function_supports_type_or_return_or_returns.js", 
  "functions": [
    {
      "desc": "a function which converts URLs to filenames", 
      "name": "toFilename", 
      "returns": {
        "desc": "A string.", 
        "type": "string"
      }, 
      "source_lines": [
        6, 
        11
      ]
    }, 
    {
      "desc": "a function which converts URLs to filenames", 
      "name": "toFilename2", 
      "returns": {
        "desc": "A String.", 
        "type": "string"
      }, 
      "source_lines": [
        12, 
        17
      ]
    }, 
    {
      "desc": "a function which converts URLs to filenames", 
      "name": "toFilename3", 
      "returns": {
        "desc": "Yeah, A String.", 
        "type": "string"
      }, 
      "source_lines": [
        18, 
        22
      ]
    }
  ], 
  "module": "function_supports_type_or_return_or_returns", 
  "source_lines": [
    1, 
    5
  ]
}
